tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/server/AbstractStructuredIdeTest.java

```diff
@@ -71,39 +75,80 @@ protected Project test(String contents, T t) {
 	 */
 	protected Project test(String moduleName, String contents, T t) {
 		moduleName = getModuleNameOrDefault(moduleName);
-		Map<String, String> srcFileNameToContents = Collections.singletonMap(moduleName, contents);
-		return test(srcFileNameToContents, moduleName, t);
+		String nameWithSelector = moduleName + MODULE_SELECTOR;
+		ArrayList<Pair<String, String>> srcFileNameToContents = Lists.newArrayList(Pair.of(nameWithSelector, contents));
+		return test(srcFileNameToContents, t);
 	}
 
-	/** Same as {@link #test(Iterable, String, Object)}, using {@link AbstractIdeTest#MODULE_NAME} as module name. */
-	protected Project test(Iterable<? extends Pair<String, String>> moduleNameToContents, T t) {
-		return test(moduleNameToContents, MODULE_NAME, t);
+	/** Same as {@link #testWS(List, Object)}, but creates a default project with name {@link #DEFAULT_PROJECT_NAME}. */
+	protected Project test(List<Pair<String, String>> modulesContents, T t) {
+		ArrayList<Pair<String, String>> modulesContentsCpy = new ArrayList<>(modulesContents);
+		modulesContentsCpy.add(Pair.of(DEPENDENCIES, N4JS_RUNTIME_NAME));
+
+		Pair<String, List<Pair<String, String>>> pairDefaultPrj = Pair.of(DEFAULT_PROJECT_NAME, modulesContentsCpy);
+		Pair<String, String> pairN4jsRuntime = Pair.of(NODE_MODULES + N4JS_RUNTIME_NAME, null);
+		ArrayList<Pair<String, List<Pair<String, String>>>> projects = new ArrayList<>();
+		projects.add(pairDefaultPrj);
+		// projects.add(pairN4jsRuntime);
+
+		modulesContentsCpy.add(pairN4jsRuntime);
+
+		return testWS(projects, t);
 	}
 
 	/**
-	 * Same as {@link #test(Map, String, Object)}, but name and content of the modules can be provided as {@link Pair
-	 * pairs}.
+	 * Same as {@link #test(LinkedHashMap, String, String, Object)}, but name and content of the modules can be provided
+	 * as {@link Pair pairs}. Finds the selected module using the {@link #MODULE_SELECTOR}.
 	 */
-	protected Project test(Iterable<? extends Pair<String, String>> moduleNameToContents, String moduleName, T t) {
-		Map<String, String> moduleNameToContentsAsMap = Streams.stream(moduleNameToContents)
-				.collect(Collectors.toMap(Pair::getKey, Pair::getValue));
-		return test(moduleNameToContentsAsMap, moduleName, t);
+	protected Project testWS(List<Pair<String, List<Pair<String, String>>>> projectsModulesContents, T t) {
```

Comment:
Could the functionality of this method (i.e. selection of a "main" module with `*`) be useful in `AbstractIdeTest`?

Reply:
Good idea, I'd like to do it in a small follow-up task along with the missing documentation (see below) and a refactoring.